### ISSUE TITLE
provide example gaf examples

### ISSUE THREAD

**Issue by jhpoelen** [jhpoelen/samara#11]
as discussed in meeting July 5, 2015 - 

@austinmeier to provide some manually constructed GAF examples for few scraped grin (and apsnet?) observations. 

With these examples, we can write unit tests to automate and test the transformation from tabular (tsv) to GAF format.

**Comment by austinmeier** [comment]
manually created example GAF2 line using the second line from grin-16k.tsv.gz:  GAF format is tab separated as well.

1140225	GRIN	PAMPA		TO:0000598	reference:xxxxxxx	EXP	from_country(Argentina)	T	PAMPA	PI 162787	germplasm	NCBITaxon:3879	07062016	austinmeier	has_phenotype_score(29.4)	

in this case I have used 'PAMPA' as the germplasm's name and symbol.  These could be switched with the PI number, which is currently in the synonyms column.

TO:0000598	was used as the trait, however this would likely be the parent of the actual trait, because TO:0000598 is 'protein content', so it refers to the protein content of a whole plant, or plant part.  The observation is the protein content of the leaf, so it would not be mapped perfectly.
it seems there are two different method IDs used for "Crude protein in the leaf."

country of origin not present (Argentina, found on this page: https://npgsweb.ars-grin.gov/gringlobal/AccessionDetail.aspx?id=1140225)

reference ID not included.  I am not sure if there is an easy way to add references, but there were two listed on the above page (D.Z. Skinner. 1999. Non random chloroplast DNA hypervariability in Medicago sativa. Theor Appl Genet Theoretical and applied genetics; international journal of b. , and D.H. Basigalup, D.K. Barnes, and R.E. Stucker. 1995. Development of a Core Collection for Perennial Medicago Plant Introductions. Crop Sci 35:1163-1168. )

The NCBI taxon used was 3879, as I did not see 300359 on NCBI

The unit on the phenotype score is unknown.  This would be nice to include, but might require new terms specific to each species/trait

**Comment by jhpoelen** [comment]
@austinmeier thanks for providing the example. For sake of completeness, the top three lines of https://github.com/jhpoelen/samara/releases/download/v0.1.2/grin-16k.tsv.gz are:

```tsv
taxon id	taxon name	descriptor id	descriptor name	descriptor definition	method id	method name	observed value	accession id	accession number	accession name
300359	Medicago sativa L. subsp. sativa	68104	Core subset (CORE)	In-vitro dry matter disappearance (ivdmd) expressed as a percent of the cultivar venal.  Higher than 100% suggests low digestibility & higher by-pass protein.	391002	ALFALFA.PROTBYPASS.93.VOLENEC	140	1140225	PI 162787	'PAMPA'
300359	Medicago sativa L. subsp. sativa	68102	Core subset (CORE)	Crude protein in the leaf.	490600	ALFALFA.1989.BARNES	29.4	1140225	PI 162787	'PAMPA'
```

**Comment by jhpoelen** [comment]
@austinmeier is there a resource you'd recommend that describes the GAF2 format?

**Comment by austinmeier** [comment]
Absolutely!  Here is the link from the Gene Ontology:  http://geneontology.org/page/go-annotation-file-format-20

It was developed to annotate genes, and gene products, so we are twisting it, and repurposing it to describe germplasm.  I have a whole slideshow walking through an example if you want me to email it over.

**Comment by jhpoelen** [comment]
@austinmeier thanks for gaf2 reference. I've created https://github.com/jhpoelen/samara/wiki to attempt to better understand the manual mapping you provided. Could you please review and comment? If it is easier to do voice /skype, please let me know. So far, I've identifier #12 #13 #14 #15 as gaps we need to fill in order to complete the samara --> gaf2 mapping . Please feel free to open new / comment on existing.
